libbeat/outputs/logstash/protocol.go

```diff
@@ -285,6 +295,49 @@ func (p *protocol) serializeDataFrame(
 	return nil
 }
 
+func serializeEvent(event common.MapStr, beat []byte) ([]byte, error) {
+	buf := bytes.NewBuffer(nil)
+	buf.WriteRune('{')
+
+	if _, hasMeta := event["@metadata"]; !hasMeta {
+		typ := event["type"].(string)
+
+		buf.WriteString(`"@metadata":{"type":`)
+		tmp, err := json.Marshal(typ)
+		if err != nil {
+			return nil, err
+		}
+		buf.Write(tmp)
+
+		buf.WriteString(`,"beat":`)
+		buf.Write(beat)
```

Comment:
Shouldn't this also be passed through `json.Marshal` or otherwise escaped? We have to be careful with JSON injection if we're doing this by hand.